### PR TITLE
#90 Follow a demo

### PR DESCRIPTION
**Pull request by mariannemyers**
Add the ability to follow a demo using Alerting.  

* New bootstrap will be required to configure fast reverse searches.
* app_specific.rb has a new function, setup_alerting, to create the Alerting config, trigger, and action for email alerts.  This must be run to configure Alerting for this issue.

Add two buttons to control following and unfollowing a demo to demo.html.  Display the proper one based on if the logged in user is following the current demo (or not).  Clicking Follow POSTs to the new follow REST endpoint and then creates and persists the Alerting rule and also saves information to the user profile.  Unfollow DELETEs the rule and removes the entry from the user's profile.

### CODE REVIEW

File: rest-api/ext/user-login.xqy [hunk 1]
```diff
@@ -2,6 +2,9 @@ xquery version "1.0-ml";
 
 module namespace user = "http://marklogic.com/rest-api/resource/user-login";
```
Comment (grtjn) [inline rest-api/ext/user-login.xqy]: user-login extension is no longer used. /user/login goes to profile extension..

File: rest-api/ext/user-status.xqy [hunk 2]
```diff
@@ -3,6 +3,8 @@ xquery version "1.0-ml";
 module namespace user = "http://marklogic.com/rest-api/resource/user-status";
```
Comment (grtjn) [inline rest-api/ext/user-status.xqy]: user-status extension is no longer used. /user/login goes to profile extension.

Maybe we should get rid of these.. :)

**Comment by grtjn** [comment]
I'll take a closer look tomorrow, unless someone else has time now..

**Comment by mariannemyers** [comment]
Added a new commit to remove changes to the user-login and user-status.  Should I delete those modules as well?

**Comment by ryanjdew** [comment]
That is fine @mariannemyers. I'll get your changes merged and remove the extensions that are no longer used in a separate commit.

**Pull request merged**